### PR TITLE
Add GitHub authentication

### PR DESCRIPTION
**Pull request by GDWR**
Adds back GitHub authentication, mirroring the Discord implementation. 
Adds documentation for setting up GitHub OAuth2 application.

### CODE REVIEW

**Comment by GDWR** [comment]
Closes #95

File: abandonauth/routers/ui.py [hunk 1]
```diff
@@ -392,3 +395,39 @@ async def discord_callback(request: Request) -> RedirectResponse:
...
+    if request_state := request.query_params.get("state"):
+        app_id, redirect_url = request_state.split(",")
+
+        dev_app = await DeveloperApplication.prisma().find_unique(
+            where={"id": app_id},
+            include={"callback_uris": True},
+        )
+
+        # This check is very important. application ID and callback URI must be validated
+        # The state in the discord login URL cannot be trusted
```
Comment (fisher60) [inline abandonauth/routers/ui.py:416]:
```suggestion
        # The state in the github login URL cannot be trusted
```

**Review by fisher60 — CHANGES_REQUESTED** [review]
Looks good, just a small mistake in docstring

**Review by fisher60 — APPROVED** [review]
LGTM

**Pull request merged**